### PR TITLE
mgr/dashboard_v2: Datatable related code improvements

### PR DESCRIPTION
**Pull request by votdev**
This PR will cleanup some datatable related issues, e.g. make use of CdTableColumn, use flex columns or remove useless code.

Additional a progress indicator is displayed during the data is loaded.

peek 2018-02-16 13-09

Signed-off-by: Volker Theile <vtheile@suse.com>

### CODE REVIEW

**Review by LenzGr — DISMISSED** [review]
lgtm, thank you

**Review by LenzGr — DISMISSED** [review]

**Comment by ricardoasmarques** [comment]
I really like this progress indicator :)

**Review by LenzGr — DISMISSED** [review]
Nice!

**Comment by ricardoasmarques** [comment]
Shouldn't we use the same "(fetchData)" approach on hosts datatable?

**Comment by votdev** [comment]
@ricardoasmarques Good spotting. Will do that.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss [hunk 1]
```diff
@@ -76,13 +76,48 @@
...
 
 ::ng-deep .oadatatable {
   border: $border-color;
   margin-bottom: 0;
   max-width: none!important;
+  .progress-linear {
+    display: block;
+    position: relative;
+    width: 100%;
+    height: 5px;
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss:91]: Could  you use a mixin for
`
    display: block;
    position: relative;
    width: 100%;
    height: 5px;
`

**Review by Devp00l — CHANGES_REQUESTED** [review]

**Review by ricardoasmarques — DISMISSED** [review]
LGTM

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss (hunk 1 above)
Comment (votdev) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss:91]: Not necessary after code cleanup.

**Review by ricardoasmarques — APPROVED** [review]
lgtm

**Review by Devp00l — APPROVED** [review]

**Pull request merged**